clients/cloud/python/producer.py

```diff
@@ -0,0 +1,104 @@
+    required = parser.add_argument_group('required arguments')
+    required.add_argument('-f', dest="config_file", help="path to Confluent Cloud configuration file", required=True)
+    required.add_argument('-t', dest="topic", help="topic name", required=True)
+    args = parser.parse_args()
+    config_file = args.config_file
+    topic = args.topic
+
+    # Read Confluent Cloud configuration for librdkafka clients
+    conf = {}
+    with open(config_file) as fh:
+      for line in fh:
+        line = line.strip()
+        if line[0] != "#" and len(line) != 0:
+          parameter, value = line.strip().split('=', 1)
+          conf[parameter] = value.strip()
+
+    # Create Producer instance
+    p = Producer({
+           'bootstrap.servers': conf['bootstrap.servers'],
+           'broker.version.fallback': '0.10.0.0',
+           'api.version.fallback.ms': 0,
+           'sasl.mechanisms': 'PLAIN',
+           'security.protocol': 'SASL_SSL',
+           'sasl.username': conf['sasl.username'],
+           'sasl.password': conf['sasl.password']
+    })
+
+    # Create topic if needed
+    # Examples of additional admin API functionality: https://github.com/confluentinc/confluent-kafka-python/blob/master/examples/adminapi.py
+    a = AdminClient({
+           'bootstrap.servers': conf['bootstrap.servers'],
+           'sasl.mechanisms': 'PLAIN',
+           'security.protocol': 'SASL_SSL',
+           'sasl.username': conf['sasl.username'],
+           'sasl.password': conf['sasl.password']
+    })
+    fs = a.create_topics([NewTopic(topic, num_partitions=1, replication_factor=3)])
+    for topic, f in fs.items():
+        try:
+            f.result()  # The result itself is None
+            print("Topic {} created".format(topic))
+        except Exception as e:
+            # Continue if error code TOPIC_ALREADY_EXISTS, which may be valid if topic exists
+            if e.args[0].code() != KafkaError.TOPIC_ALREADY_EXISTS:
+              print("Failed to create topic {}: {}".format(topic, e))
+
+    # Optional per-message on_delivery handler (triggered by poll() or flush())
+    # when a message has been successfully delivered or
+    # permanently failed delivery (after retries).
+    def acked(err, msg):
+        """Delivery report handler called on successful or failed delivery of the message."""
+        if err is not None:
+            print("failed to deliver message: {0}".format(err.str()))
+        else:
+            print("Successfully produced record to topic {} partition [{}] @ offset {}".format(msg.topic(), msg.partition(), msg.offset()))
+
+    for n in range(10):
+      record_key = "alice"
+      record_value = json.dumps({'count': n})
+      print ("Preparing to produce record: {} \t {}".format(record_key, record_value))
+      p.produce(topic,key=record_key,value=record_value,on_delivery=acked)
+      # p.poll(0) serves delivery reports from previous produce() calls to the on_delivery handler
```

Comment:
This doesn't read that well in my opinion

`# p.poll(0) serves delivery reports (on_delivery) from previous produce() calls.`

Reply:
Ack